### PR TITLE
Migrate to spring boot V3 (#3969)

### PR DESCRIPTION
**Pull request by vlo-rte**
I think today we can not migrate to spring boot 3 because spring boot 3 uses jakarta servlet-api 6.0 and jetty does not yet integrate servlet-api 6.0. To make opfab works we have to downgrade servlet-api to version 5.0 but with this downgrade, our java tests fail because they need servlet-api 6.0.

With servlet-api 6.0 (and so java tests successful), we have this error when we start opfab : 

```org.springframework.context.ApplicationContextException: Unable to start web server
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:164) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:578) ~[spring-context-6.0.3.jar!/:6.0.3]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:432) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:308) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1291) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.opfab.users.UsersApplication.main(UsersApplication.java:34) ~[classes!/:SNAPSHOT]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[users-service-SNAPSHOT.jar:SNAPSHOT]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:95) ~[users-service-SNAPSHOT.jar:SNAPSHOT]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[users-service-SNAPSHOT.jar:SNAPSHOT]
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65) ~[users-service-SNAPSHOT.jar:SNAPSHOT]
Caused by: java.lang.NoClassDefFoundError: jakarta/servlet/http/HttpSessionContext
	at org.eclipse.jetty.servlet.ServletContextHandler.newSessionHandler(ServletContextHandler.java:339) ~[jetty-servlet-11.0.13.jar!/:11.0.13]
	at org.eclipse.jetty.servlet.ServletContextHandler.getSessionHandler(ServletContextHandler.java:432) ~[jetty-servlet-11.0.13.jar!/:11.0.13]
	at org.eclipse.jetty.servlet.ServletContextHandler.relinkHandlers(ServletContextHandler.java:257) ~[jetty-servlet-11.0.13.jar!/:11.0.13]
	at org.eclipse.jetty.servlet.ServletContextHandler.<init>(ServletContextHandler.java:180) ~[jetty-servlet-11.0.13.jar!/:11.0.13]
	at org.eclipse.jetty.webapp.WebAppContext.<init>(WebAppContext.java:301) ~[jetty-webapp-11.0.13.jar!/:11.0.13]
	at org.eclipse.jetty.webapp.WebAppContext.<init>(WebAppContext.java:228) ~[jetty-webapp-11.0.13.jar!/:11.0.13]
	at org.springframework.boot.web.embedded.jetty.JettyEmbeddedWebAppContext.<init>(JettyEmbeddedWebAppContext.java:28) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.boot.web.embedded.jetty.JettyServletWebServerFactory.getWebServer(JettyServletWebServerFactory.java:158) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:183) ~[spring-boot-3.0.1.jar!/:3.0.1]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:161) ~[spring-boot-3.0.1.jar!/:3.0.1]
	... 16 common frames omitted
Caused by: java.lang.ClassNotFoundException: jakarta.servlet.http.HttpSessionContext
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:587) ~[na:na]
	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:149) ~[users-service-SNAPSHOT.jar:SNAPSHOT]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[na:na]
	... 26 common frames omitted```

### CODE REVIEW

**Comment by freddidierRTE** [comment]
Replace jetty by undertow solve the pb

File: src/docs/asciidoc/resources/migration_guide_to_3.12.adoc [hunk 1]
```diff
@@ -46,6 +46,14 @@ usercard configuration in the config.json of your bundle.
...
\ No newline at end of file
+The deprecated method 'templateGateway.getSpecificCardInformation()' in usercard templates has been removed, use 'usercardTemplateGateway.getSpecificCardInformation()' method instead.
+
+
+== Trailing slash
+
+When calling an opfab API you cannot use anymore trailing slash on url.
+
+
+If, for example, one makes a request like https://opfab/businessconfig/processes/, it must be replace by https://opfab/businessconfig/processes
```
Comment (vlo-rte) [inline src/docs/asciidoc/resources/migration_guide_to_3.12.adoc:59]: Please put this text in a file for migration to 3.13.0

File: src/test/api/karate/businessconfig/security.feature [hunk 2]
```diff
@@ -6,20 +6,47 @@ Feature: Security
...
+# Check template v2 
+Given url opfabUrl + '/businessconfig/processes/'+ process +'/templates/template?version=2'
 And header Authorization = 'Bearer ' + authTokenAsTSO
 When method GET
-Then status 401
+Then status 200
+And match response != ""
+
+
+# Try to access version 1 file form version 2 via path traversal
```
Comment (vlo-rte) [inline src/test/api/karate/businessconfig/security.feature:31]: from ?

File: src/test/api/karate/businessconfig/security.feature [hunk 3]
```diff
@@ -6,20 +6,47 @@ Feature: Security
...
+# actual spring implementation return a code 200 wth no content when inserting ../ in url
+Given url opfabUrl + '/businessconfig/processes/'+ process +'/templates/../../1/templates/template?version=2'
+And header Authorization = 'Bearer ' + authTokenAsTSO
+When method GET
+Then status 200
+And match response == ""
+
+# Try to access version 1 file form version 2 via path traversal
```
Comment (vlo-rte) [inline src/test/api/karate/businessconfig/security.feature:39]: from ?

File: src/test/api/karate/businessconfig/security.feature [hunk 4]
```diff
@@ -6,20 +6,47 @@ Feature: Security
...
+# actual spring implementation return a code 200 wth no content when inserting encoding / (%2F)  in url
```
Comment (vlo-rte) [inline src/test/api/karate/businessconfig/security.feature:40]: returns

**Pull request merged**